testcases/OpTestInterrupt.py

```diff
@@ -0,0 +1,189 @@
+            self.con.run_command("grub2-mkconfig -o /boot/grub2/grub.cfg")
+            self.con.run_command("sync;sleep 10")
+            self.cv_SYSTEM.goto_state(OpSystemState.OFF)
+            self.cv_SYSTEM.goto_state(OpSystemState.OS)
+            self.con.run_command("cat /proc/cmdline")
+            self.con.run_command_ignore_fail("cat /proc/interrupts | grep -Eai 'xive'")
+            self.con.run_command("sed -i 's/xive=.*/xive=on\"/' /etc/default/grub")
+            self.con.run_command("grub2-mkconfig -o /boot/grub2/grub.cfg")
+            self.con.run_command("sync;sleep 10")
+            self.cv_SYSTEM.goto_state(OpSystemState.OFF)
+            self.cv_SYSTEM.goto_state(OpSystemState.OS)
+            self.con.run_command("cat /proc/cmdline")
+        elif 'Red Hat' in res[0].strip():
+            self.distro = 'RHEL'
+            self.rhel_grub('off')
+            self.rhel_grub('on')
+
+    def rhel_grub(self, param):
+        '''
+        This function will pass xive=on and off parameters
+        to command line.
+        '''
+        self.con.run_command("grubby --info=/boot/vmlinuz-`uname -r`")
+        self.con.run_command_ignore_fail("grubby --remove-args=xive* --update-kernel=/boot/vmlinuz-`uname -r`")
+        self.con.run_command("grubby --args=xive=%s --update-kernel=/boot/vmlinuz-`uname -r`" % param)
+        self.con.run_command("sync; sync; sleep 5")
+        self.cv_SYSTEM.goto_state(OpSystemState.OFF)
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        self.con.run_command("grubby --info=/boot/vmlinuz-`uname -r`")
+
+    def interrupt_cpu_check(self): 
+        '''
+        This function will validate interrupts handled by CPU's.
+        Make sure while running this function xive should be on
+        '''
+        self.con = self.cv_SYSTEM.cv_HOST.get_ssh_connection()
+        res = self.con.run_command("ip route list | grep default | awk '{print $5}'")
+        inter = "cat /proc/interrupts | grep %s | head -1" % res[0]
+        self.con.run_command(inter)
+        self.con.run_command("ppc64_cpu --cores-on")
+        self.con.run_command("ppc64_cpu --cores-on=1")
+        self.con.run_command(inter)
+        temp = "%s | awk '{print $2}'" % inter
+        self.con.run_command("ppc64_cpu --smt=off")
+        tmp1 = self.con.run_command(temp)
+        time.sleep(10)
+        tmp2 = self.con.run_command(temp)
+        if (tmp1[0] < tmp2[0]):
+            log.info("Interrupts are handled by CPU0")
+            self.con.run_command("ppc64_cpu --cores-on=all")
+            self.con.run_command("ppc64_cpu --smt=on")
+            output = self.con.run_command(inter)
+            cmd = "%s | awk '{print $4}'" % inter
+            tmp3 = self.con.run_command(cmd)
+            smp = "%s | awk '{print $1}'" % inter
+            result = self.con.run_command(smp)
+            self.con.run_command("cd /proc/irq/%s" % result[0].lstrip().split(':')[0])
+            self.con.run_command("cat smp_affinity_list")
```

Comment:
Can't you use absolute path and run `cat` instead of switching to that directory?